pkg/gui.py

```
@@ -348,3 +344,89 @@ class SettingsMenu(OtherFrame):
+        self.debug_level_options = ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
+        self.debug_level_menu = OptionMenu(self, self.debug_level_var, *self.debug_level_options)
+        self.debug_level_menu.grid(row=3, column=1, sticky=W+E)
+
+        # Load settings from the JSON file
+        self.load_settings()
+
+        # Create a save button
+        self.save_btn = Tk.Button(self, text="Save", command=self.save_settings)
+        self.save_btn.grid(row=4, column=0, columnspan=2, sticky=W+E+N)
+
+    def update_settings(self, settings) -> None:
```

Ruff (0.8.0)

Missing type annotation for function argument `settings`

(ANN001)